### ISSUE TITLE
LL ADC has duplicated arguments

### ISSUE THREAD

**Issue by vpetrigo** [STMicroelectronics/STM32CubeL4#31]
Hello,

This part in the ADC LL driver contains duplicated arguments in `&` chain:

https://github.com/STMicroelectronics/STM32CubeL4/blob/d023c0d560ace11509f9b761c8913a9e48fcf194/Drivers/STM32L4xx_HAL_Driver/Inc/stm32l4xx_ll_adc.h#L5216

Since `READ_BIT` is expanded to that:

```c
((*preg) & AWDy))
```

The second `AWDy` usage is redundant. So all that expression can be simplified to that:

```c
uint32_t AnalogWDMonitChannels = (READ_BIT(*preg, AWDy) & ADC_AWD_CR_ALL_CHANNEL_MASK);
```

**Comment by ASELSTM** [comment]
Hi @vpetrigo,

The issue you pointed out has been confirmed. Indeed, The second `AWDy `usage is redundant.
A fix will be implemented and made available in future release. Thank you once again for your contribution.

With regards,

**Comment by ASELSTM** [comment]
ST Internal Reference: 101705

**Comment by ASELSTM** [comment]
Hi @vpetrigo,

Thank you for your contribution. This issue has been fixed in the frame of version v1.17.1 of the STM32CubeL4. Please allow me then to close this thread.

With regards,
